builtin/providers/aws/resource_aws_elasticache.go

```diff
@@ -0,0 +1,256 @@
+		securityNames[i] = name.(string)
+	}
+	securityIds := make([]string, securityIdSet.Len())
+	for i, id := range securityIdSet.List() {
+		securityIds[i] = id.(string)
+	}
+
+	req := &elasticcache.CreateCacheClusterMessage{
+		CacheClusterID:          aws.String(clusterId),
+		CacheNodeType:           aws.String(nodeType),
+		NumCacheNodes:           aws.Integer(numNodes),
+		Engine:                  aws.String(engine),
+		EngineVersion:           aws.String(engineVersion),
+		Port:                    aws.Integer(port),
+		CacheSubnetGroupName:    aws.String(subnetGroupName),
+		CacheSecurityGroupNames: securityNames,
+		SecurityGroupIDs:        securityIds,
+		CacheParameterGroupName: aws.String(paramGroupName),
+	}
+
+	_, err := elasticacheconn.CreateCacheCluster(req)
+	if err != nil {
+		return fmt.Errorf("Error creating Elasticache: %s", err)
+	}
+
+	d.SetId(clusterId)
+
+	return nil
+}
+
+func resourceAwsElasticacheRead(d *schema.ResourceData, meta interface{}) error {
+	elasticacheconn := meta.(*AWSClient).elasticacheconn
+	req := &elasticcache.DescribeCacheClustersMessage{
+		CacheClusterID: aws.String(d.Id()),
+	}
+
+	res, err := elasticacheconn.DescribeCacheClusters(req)
+	if err != nil {
+		return err
+	}
+
+	if len(res.CacheClusters) == 1 {
+		c := res.CacheClusters[0]
+		d.Set("cluster_id", c.CacheClusterID)
+		d.Set("node_type", c.CacheNodeType)
+		d.Set("num_cache_nodes", c.NumCacheNodes)
+		d.Set("engine", c.Engine)
+		d.Set("engine_version", c.EngineVersion)
+		if c.ConfigurationEndpoint != nil {
+			d.Set("port", c.ConfigurationEndpoint.Port)
+		}
+		d.Set("subnet_group_name", c.CacheSubnetGroupName)
+		d.Set("security_group_names", c.CacheSecurityGroups)
+		d.Set("security_group_ids", c.SecurityGroups)
+		d.Set("parameter_group_name", c.CacheParameterGroup)
+	}
+
+	return nil
+}
+
+func resourceAwsElasticacheUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceAwsElasticacheDelete(d *schema.ResourceData, meta interface{}) error {
+	elasticacheconn := meta.(*AWSClient).elasticacheconn
+
+	// AWS CLI complains like below if we try to delete while creating.
+	//     Can only delete cache clusters with state in:
+	//     available, failed, incompatible-parameters, incompatible-network, restore-failed
+	// So we cannot delete clusters in `creating` state.
+	// We need to wait for states which we can delete.
+	pending := []string{"creating"}
+	stateConf := &resource.StateChangeConf{
+		Pending:    pending,
+		Target:     "available",
+		Refresh:    CacheClusterStateRefreshFunc(elasticacheconn, d.Id(), "available", pending),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	log.Printf("[DEBUG] Waiting for state to become available: %v", d.Id())
+	_, sterr := stateConf.WaitForState()
+	if sterr != nil {
+		return fmt.Errorf("Error waiting for elasticache (%s) to be deletable: %s", d.Id(), sterr)
+	}
```

Comment:
Wouldn't it make more sense to move this polling into the end of `resourceAwsElasticacheCreate` so that by the time we return from that we can be sure that the cluster is ready to go?